commitserver/commit/mkdir_provider.go

```diff
@@ -0,0 +1,10 @@
+package commit
+
+import (
+	"os"
+)
+
+// MkdirAllProvider is an interface for creating directories.
+type MkdirAllProvider interface {
+	MkdirAll(path string, perm os.FileMode) error
```

Comment:
I think the interface we need is `func MkdirAll(root, unsafePath string, mode int) error`. The Linux implementation will just pass through the args to `securejoin.MkdirAll`. The non-Linux implementation will run `securejoin.SecureJoin(root, unsafePath)` and then run `os.MkdirAll`.

Reply:
Then, is it correct that you want to treat the part you uploaded on the issue with func MkdirAll (root, unsafe path string, mode int) error??
 
before

``` go
fullHydratePath, err = securejoin.SecureJoin(h.dirPath, hydratePath) 
 if err != nil { 
 	return fmt.Errorf("failed to construct hydrate path: %w", err) 
 } 
 // TODO: consider switching to securejoin.MkdirAll: https://github.com/cyphar/filepath-securejoin?tab=readme-ov-file#mkdirall 
 err = os.MkdirAll(fullHydratePath, os.ModePerm) 
```

after
``` go
var fullHydratePath string
err = MkdirAll(rootPath, hydratePath, os.ModePerm)
```